Slim/Plugin/Podcast/Parser.pm

```diff
@@ -66,42 +69,47 @@ sub parse {
 		}
 
 		$cache->set("$key-duration", $item->{duration}, '30days');
+
+		my $url = $item->{enclosure}->{url};
+		$item->{enclosure}->{url} = Slim::Plugin::Podcast::Plugin::wrap($url);
 		
 		# if we've played this podcast before, add a menu level to ask whether to continue or start from scratch
-		if ( $position && $position < $item->{duration} - 15 ) {
-			delete $item->{description};     # remove description, or xmlbrowser would consider this to be a RSS feed
-
-			my $enclosure = delete $item->{enclosure};
-			$position     = $cache->get('podcast-' . $enclosure->{url});
-
-			$position = Slim::Utils::DateTime::timeFormat($position);
+		if ( $from && $from < $item->{duration} - 15 ) {
+			$position = Slim::Utils::DateTime::timeFormat($from);
 			$position =~ s/^0+[:\.]//;
-
+			
+			# remote_image is now cached, so replace enclosure with a play attribute 
+			# so that XMLBrowser to show sub-menu *and* we can play from top
+			my $enclosure = delete $item->{enclosure};
+			$item->{on_select} = 'play';
+			$item->{play} = Slim::Plugin::Podcast::Plugin::wrap($url);
+			$item->{type}  = 'link';
+			
 			$item->{items} = [{
 				title => cstring($client, 'PLUGIN_PODCAST_PLAY_FROM_POSITION_X', $position),
 				name  => cstring($client, 'PLUGIN_PODCAST_PLAY_FROM_POSITION_X', $position),
+				description => $item->{description},
+				cover => $item->{image},
```

Comment:
Where's the image defined? You mention `remote_image` above, but it's not used here, is it?

Reply:
It's done by the OPML browser that knows how to handle RSS feeds and break them appart. I don't to anything here

Reply:
ok